### PR TITLE
Improved destructuring with default value assign

### PR DESCRIPTION
**Pull request by fsgreco**
#### Summary

I'm trying to improve the explanation on how to de-structure parameters and assign default values on the function arguments.

#### Motivation

I found the explanation that is published nowadays hard to understand.

This PR…
-->
- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/functions/default_parameters/index.md [hunk 1]
```diff
@@ -233,14 +233,30 @@ f(2)  // [2, undefined]
 
 You can use default value assignment with the
 {{jsxref("Operators/Destructuring_assignment", "destructuring assignment", "", 1)}}
-notation:
+notation. 
+
+A common way of doing it consist in destructuring and assigning the values, and then assign an empty object/array to that destructured object, e.g:`{z = 3} = {}` .
```
Comment (tshemsedinov) [inline files/en-us/web/javascript/reference/functions/default_parameters/index.md:238]:
```suggestion
A common way of doing it consist in destructuring and assigning the values, and then assign an empty object/array to that destructured object, e.g:`{ z = 3 } = {}` . 
```

**Review by tshemsedinov — COMMENTED** [review]
Just a few code style proposals

**Comment by anonimoconiglio** [comment]
Thanks @tshemsedinov I just added your suggestions!

File: files/en-us/web/javascript/reference/functions/default_parameters/index.md [hunk 2]
```diff
@@ -233,14 +233,30 @@ f(2)  // [2, undefined]
 
 You can use default value assignment with the
 {{jsxref("Operators/Destructuring_assignment", "destructuring assignment", "", 1)}}
-notation:
+notation. 
+
+A common way of doing it consist in destructuring and assigning the values, and then assign an empty object/array to that destructured object, e.g:`{ z = 3 } = {}` .
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/functions/default_parameters/index.md:238]:
```suggestion
A common way of doing that is to assign an empty object/array to the destructured object/array, e.g:`{ z = 3 } = {}` . 
```

File: files/en-us/web/javascript/reference/functions/default_parameters/index.md [hunk 3]
```diff
@@ -233,14 +233,30 @@ f(2)  // [2, undefined]
 
 You can use default value assignment with the
 {{jsxref("Operators/Destructuring_assignment", "destructuring assignment", "", 1)}}
-notation:
+notation. 
+
+A common way of doing it consist in destructuring and assigning the values, and then assign an empty object/array to that destructured object, e.g:`{ z = 3 } = {}` . 
+By doing it this way it is possible to pass an empty array/object to the function and still have those values prefilled:
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/functions/default_parameters/index.md:239]:
```suggestion
By doing it that way, you make it possible to pass an empty array/object to the function and still have those values prefilled:
```

File: files/en-us/web/javascript/reference/functions/default_parameters/index.md [hunk 4]
```diff
@@ -233,14 +233,30 @@ f(2)  // [2, undefined]
...
+
+preArray();      // 3
+preArray([]);    // 3
+preArray([2]);   // 4
+preArray([2, 3]); // 5
+
+
+// Works the same for Objects:
+function preObj({z = 3} = {}) {
+  return z;
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/functions/default_parameters/index.md:254]:
```suggestion
function preFilledArray([x = 1, y = 2] = []) {
  return x + y;
}

preFilledArray();      // 3
preFilledArray([]);    // 3
preFilledArray([2]);   // 4
preFilledArray([2, 3]); // 5

// Works the same for objects:
function preFilledObject({z = 3} = {}) {
  return z;
```

File: files/en-us/web/javascript/reference/functions/default_parameters/index.md [hunk 5]
```diff
@@ -233,14 +233,29 @@ f(2)  // [2, undefined]
...
+
+// Works the same for objects:
+function preFilledObject({z = 3} = {}) {
+  return z;
 }
 
-f()  // 6
+preObj();          // 3
+preObj({});        // 3
+preObj({ z: 2 });  // 2
```
Comment (sideshowbarker) [inline files/en-us/web/javascript/reference/functions/default_parameters/index.md:258]:
```suggestion
preFilledObject();          // 3
preFilledObject({});        // 3
preFilledObject({ z: 2 });  // 2
```

**Review by sideshowbarker — APPROVED** [review]

**Comment by sideshowbarker** [comment]
Santiago, thanks much, and congrats on landing your first docs change here — welcome aboard 🎉

**Pull request merged**